pyhstr/application.py

```diff
@@ -31,22 +31,14 @@ def search(self):
         self.user_interface.page.selected.value = 0
         self.user_interface.page.value = 1
 
-        if not self.regex_match:
-            if self.case_sensitivity:
-                self.all_entries[self.view] = [
-                    cmd for cmd in self.all_entries[self.view]
-                    if self.user_interface.search_string in cmd
-                ]
-            else:
-                self.all_entries[self.view] = [
-                    cmd for cmd in self.all_entries[self.view]
-                    if self.user_interface.search_string.lower() in cmd.lower()
-                ]
-        else:
+        try:
             self.all_entries[self.view] = [
                 cmd for cmd in self.all_entries[self.view]
-                if re.search(self.user_interface.search_string, cmd)
+                if self.user_interface.create_search_string_regex().search(self.user_interface.search_string)
```

Comment:
I was wondering why it didn't seem to filter anything. Here's why.

```suggestion
                if self.user_interface.create_search_string_regex().search(cmd)
```